### PR TITLE
Postgres/MySQL/MSSQL: Change time field to "Time" for time series queries

### LINKED ISSUE #36059: SQL: Name of time field changed in v8
**What happened**:

After upgrade to 8.0.3 from 7.5.5 graph panel no longer could display data from query.

**What you expected to happen**:

Graph panel should work as before.

**How to reproduce it (as minimally and precisely as possible)**:

Upgrade from 7.5.5, use transform: filter by name

**Anything else we need to know?**:

There is a transform on my panel that filters by name, in 7.5.5 it picks out 'Time' from the $__timeGroupAlias but on upgrade to 8.0.3 'Time' is no longer available because there is a capitailsation of the 'time' field in version 7.5.5 which does not appear to happen in 8.0.3.

Both versions 7.5.5 and 8.0.3 expand the $__timeGroupAlias to time_bucket('10.0s',logtime) AS "time"

A fix is to go back through and change the name filters from 'Time' to 'time' but is quite tedious!

**Environment**:
- Grafana version:

7.5.5, 8.0.3

- Data source type & version:

timescaledb pg12

- OS Grafana is installed on:

archlinux, docker

- User OS & Browser:

Opera, win10

- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by marefr**
**What this PR does / why we need it**:
Name of time field changed in v8 for time series queries from `Time` to the name of the selected time column, i.e. `time` or `time_sec`. These changes should make sure that the name of time field is always returned as `Time` for time series queries.

**Which issue(s) this PR fixes**:
Fixes #36059

**Special notes for your reviewer**:

### CODE REVIEW

**Review by idafurjes — APPROVED** [review]
LGTM

File: pkg/tsdb/sqleng/sql_engine.go [hunk 1]
```diff
@@ -283,6 +283,10 @@ func (e *dataPlugin) executeQuery(query plugins.DataSubQuery, wg *sync.WaitGroup
 			errAppendDebug("db has no time column", errors.New("no time column found"), interpolatedQuery)
 			return
 		}
+
+		// Make sure to name the time field 'Time' to be backward compatible with Grafana pre-v8.
+		frame.Fields[qm.timeIndex].Name = "Time"
```
Comment (ryantxu) [inline pkg/tsdb/sqleng/sql_engine.go:288]:
```suggestion
		frame.Fields[qm.timeIndex].Name = data.TimeSeriesTimeFieldName
```
maybe use the constant so it is more clear that the name is meaningful?

Comment (marefr) [reply pkg/tsdb/sqleng/sql_engine.go:288]: Thanks :+1:

File: pkg/tsdb/mysql/mysql_test.go [hunk 2]
```diff
@@ -373,6 +373,7 @@ func TestMySQL(t *testing.T) {
 
 			frames, _ := queryResult.Dataframes.Decoded()
 			require.Len(t, frames, 1)
+			require.Equal(t, "Time", frames[0].Fields[0].Name)
```
Comment (marefr) [inline pkg/tsdb/mysql/mysql_test.go:376]:
```suggestion
			require.Equal(t, data.TimeSeriesTimeFieldName, frames[0].Fields[0].Name)
```

File: pkg/tsdb/mysql/mysql_test.go [hunk 3]
```diff
@@ -505,6 +506,7 @@ func TestMySQL(t *testing.T) {
 			frames, err := queryResult.Dataframes.Decoded()
 			require.NoError(t, err)
 			require.Len(t, frames, 1)
+			require.Equal(t, "Time", frames[0].Fields[0].Name)
```
Comment (marefr) [inline pkg/tsdb/mysql/mysql_test.go:509]:
```suggestion
			require.Equal(t, data.TimeSeriesTimeFieldName, frames[0].Fields[0].Name)
```

**Review by torkelo — APPROVED** [review]

**Pull request merged**